test/integration/testPutGet.js

```diff
@@ -857,3 +861,315 @@ describe('PUT GET test multiple files', function () {
+      fs.closeSync(fh);
+      resolve(true);
+    });
+  };
+
+  // Create a file of 100 MB
+  createEmptyFileOfSize(largeFile.name, 100*1024*1024);
+
+  const zeroByteFilePath = getPlatformTmpPath(zeroByteFile.name);
+  const largeFilePath = getPlatformTmpPath(largeFile.name);
+
+  const tmpdirPath = getPlatformTmpPath(tmpDir);
+  const getQuery = `GET ${stage} ${tmpdirPath}`;
+
+  const testCases =
+    [
+      {
+        name: 'zero byte file',
+        putQuery: `PUT ${zeroByteFilePath} ${stage} AUTO_COMPRESS=FALSE`,
+        size: 0
+      },
+      {
+        name: 'large size file',
+        putQuery: `PUT ${largeFilePath} ${stage} AUTO_COMPRESS=FALSE`,
+        size: 100*1024*1024
+      }
+    ];
+
+  const createItCallback = function (testCase) {
+    return function (done) {
+      {
+        async.series(
+          [
+            function (callback) {
+              connection.execute({
+                sqlText: testCase.putQuery,
+                complete: function (err, stmt) {
+                  if(err){
+                    callback(err);
+                  }else {
+                    const stream = stmt.streamRows();
+                    stream.on('error', function (err) {
+                      callback(err);
+                    });
+                    stream.on('data', function (row) {
+                      // Check the file is correctly uploaded
+                      assert.strictEqual(row['status'], UPLOADED);
+                      assert.strictEqual(row.targetSize, testCase.size);
+                    });
+                    stream.on('end', function () {
+                      callback();
+                    });
+                  }
+                }
+              });
+            },
+            function (callback) {
+              connection.execute({
+                sqlText: getQuery,
+                complete: function (err, stmt) {
+                  if(err){
+                    callback(err);
+                  }else {
+                    const stream = stmt.streamRows();
+                    stream.on('error', function (err) {
+                      callback(err);
+                    });
+                    stream.on('data', function (row) {
+                      assert.strictEqual(row['status'], DOWNLOADED);
+                      // Verify the size of the file
+                      const file = path.join(tmpDir, row.file);
+                      const size = fs.statSync(file).size;
+                      assert.strictEqual(size, testCase.size);
+                    });
+                    stream.on('end', function () {
+                      callback();
+                    });
+                  }
+                }
+              });
+            }
+          ],
+          done
+        );
+      }
+    };
+  };
+
+  for (let index = 0; index < testCases.length; index++) {
+    const testCase = testCases[index];
+    it(testCase.name, createItCallback(testCase));
```

Comment:
can we just inline `createItCallback` function here?

Reply:
it is gone now.